scripts/parallel-tsc/tsc.js

```diff
@@ -126,11 +126,11 @@ class Pool {
     taskProcess.once("exit", async code => {
       const startTime = this.#running.get(taskProcess);
       const duration = performance.now() - startTime;
-      this.#running.delete(taskProcess);
       await this.#onTaskDone(tasks, code, duration);
+      this.#running.delete(taskProcess);
 
-      // Spin  the event loop so that #run can pick up the next task
-      await setTimeout(() => {}, 0);
+      // Spin the event loop so that #run can pick up the next task
+      await 0;
```

Comment:
This doesn't actually spin the event loop. `setTimeout` has lower priority than `queueMicrotask`, while `promise.then` for an already-resolved promise has the same priority.

Reply:
Ah, I see.
Perhaps you meant `await new Promise(resolve => setTimeout(resolve, 0));`?

Reply:
Oh, yes :sweat_smile: 

Or `import { setTimeout } from "node:timers/promises"`